src/pymmcore_widgets/_mda/_positions_table_widget.py

```diff
@@ -213,88 +267,217 @@ def _add_position_row(self) -> int:
-        # data(self.POS_ROLE) is used to keep track of grid and/or position
-        # even when the user changed the name in the table.
-        item.setData(self.POS_ROLE, table_item)
-        item.setToolTip(table_item)
-        item.setTextAlignment(AlignCenter)
-        self._table.setItem(row, col, item)
+    def _add_grid_buttons(self, row: int | None, col: int | None) -> None:
+        wdg = QWidget()
+        layout = QHBoxLayout()
+        layout.setSpacing(3)
+        layout.setContentsMargins(0, 0, 0, 0)
+        wdg.setLayout(layout)
+        add_grid = QPushButton()
+        add_grid.setIcon(icon(MDI6.plus_thick, color=(0, 255, 0)))
+        add_grid.setIconSize(QSize(25, 25))
+        add_grid.setFixedHeight(25)
+        add_grid.setSizePolicy(QSizePolicy.Policy.Fixed, QSizePolicy.Policy.Fixed)
+        add_grid.setContextMenuPolicy(Qt.CustomContextMenu)
+        # for righ-click menu
+        add_grid.customContextMenuRequested.connect(self._show_apply_to_all_menu)
+        add_grid.clicked.connect(self._grid_widget)
+        remove_grid = QPushButton()
+        remove_grid.setIcon(icon(MDI6.close_thick, color="magenta"))
+        remove_grid.setIconSize(QSize(25, 25))
+        remove_grid.setFixedHeight(25)
+        remove_grid.setSizePolicy(QSizePolicy.Policy.Fixed, QSizePolicy.Policy.Fixed)
+        remove_grid.clicked.connect(self._remove_grid_plan)
+        layout.addWidget(add_grid)
+        layout.addWidget(remove_grid)
+        remove_grid.hide()
+        self._table.setCellWidget(row, col, wdg)
+
+    def _remove_grid_plan(self) -> None:
+        row = self._table.indexAt(self.sender().parent().pos()).row()
+        self._table.item(row, 0).setData(self.GRID_ROLE, None)
+        self._table.item(row, 0).setToolTip("")
+        add_grid, remove_grid = self._get_grid_buttons(row)
+        add_grid.setText("")
+        add_grid.setIcon(icon(MDI6.plus_thick, color=(0, 255, 0)))
+        add_grid.setIconSize(QSize(25, 25))
+        remove_grid.hide()
+        self._enable_button()
+        self.valueChanged.emit()
 
-    def _remove_position(self) -> None:
-        rows = {r.row() for r in self._table.selectedIndexes()}
-        grid_to_delete = []
-
-        for idx in sorted(rows, reverse=True):
-            pos_role = self._table.item(idx, 0).data(self.POS_ROLE)
-            # store grid name if is a grid position
-            if "Grid" in pos_role:
-                grid_name = pos_role.split("_")[0]
-                grid_to_delete.append(grid_name)
-            else:
-                # remove if is a single position
-                self._table.removeRow(idx)
+    def _get_grid_buttons(self, row: int) -> tuple[QPushButton, QPushButton]:
+        return (
+            self._table.cellWidget(row, 4).layout().itemAt(0).widget(),
+            self._table.cellWidget(row, 4).layout().itemAt(1).widget(),
+        )
+
+    def _grid_widget(self) -> None:
+        if not self._mmc.getXYStageDevice():
```

Comment:
this feels a bit like a bleeding of concerns.  You are making it so that this widget can *never* be used to setup a grid plan when there isn't an active stage device.  But that feels like the kind of thing that an application should do (something like napari micromanager) not something that a library of widgets should be doing